projects/vdk-core/plugins/vdk-plugin-control-cli/src/vdk/plugin/control_cli_plugin/execution_skip.py

```diff
@@ -0,0 +1,164 @@
+        :return: bool
+                 If the data job has a different execution running in K8S.
+                 A running different execution means that there is a data job execution currently
+                 submitted or running with a different id and in that case we should skip current execution
+                 until the other finishes.
+        """
+        jobs = self._get_running_job_executions(job_name, job_team)
+        job_execution_running = False
+        for job_execution in jobs:
+            # We break only if we have a job with a different execution id
+            # Sometimes ids returned by VDK execution_state can slightly differ than
+            # ids returned by the API. VDK ids  have '-xxxxx' appended at the end,
+            # where x is a random alphameric char. Otherwise ids start the same.
+            if not job_execution_attempt_id.startswith(job_execution.id):
+                log.info(
+                    f"Found another running job execution for {job_name}. Current execution id: {job_execution_attempt_id}. Other running execution id: {job_execution.id}"
+                )
+                job_execution_running = True
+                break
+
+        return job_execution_running
+
+    def _get_running_job_executions(
+        self, job_name, team_name
+    ) -> List[DataJobExecution]:
+        job_status = [
+            "submitted",
+            "running",
+        ]  # We query only for running or manual jobs
+        response = self.execution_api_client.data_job_execution_list(
+            job_name=job_name, team_name=team_name, execution_status=job_status
+        )
+        log.info(f"Api call returned job execution list: {response}")
+        return response
+
+    @staticmethod
+    def _authenticate(api_token_authorization_url, api_token, auth_type) -> None:
+        log.info("Authenticating.")
+        auth = Authentication()
+        auth.update_api_token_authorization_url(api_token_authorization_url)
+        auth.update_api_token(api_token)
+        auth.update_auth_type(auth_type)
+        auth.acquire_and_cache_access_token()
+        log.info("Login successful.")
+
+
+class ExitWithMessage:
+    def __init__(self) -> None:
+        super.__init__(self)
+
+    @staticmethod
+    def skip_job_run(job_name) -> None:
+        log.info(f"Data job: {job_name} is already running in K8S.")
+        log.info(
+            f"Skipping execution and exiting. You can retry the job when it finishes executing in K8S."
+        )
+        os._exit(0)
+
+
+def _skip_job_if_necessary(
+    log_config: str,
+    job_name: str,
+    execution_id: str,
+    job_team: str,
+    action: WriteToFileAction,
+):
+    try:
+        log.info("Checking if job should be skipped:")
+        log.info(
+            f"Job : {job_name}, Team : {job_team}, Log config: {log_config}, execution_id: {execution_id}"
+        )
+        vdk_config = VDKConfig()
+        # vdk_config.
+        if log_config != "CLOUD":
```

Comment:
I think we can proceed as is for now and fix all of these cases in one PR next sprint.

Reply:
I agree with you @tozka , however I think another PR will introduce scope creep and the execution API will be delayed even further. IMO if the proposed solution works we should use it. Nevertheless, if you think that this isn't a good pattern and we cannot afford to use it as is, I can open another PR for that. Otherwise, I can create a ticket and refactor this in the next sprint. Let me know how to proceed.

Reply:
I think next spring is OK.

Reply:
Created issue for next spring:
https://jira.eng.vmware.com/browse/SC-996